src/main/java/io/jenkins/plugins/sprp/YAML_FlowDefinition.java

```diff
@@ -101,8 +103,12 @@ public FlowExecution create(FlowExecutionOwner owner, TaskListener listener,
         this.gitConfig.setGitBranch(property.getBranch().getName());
 
         if(gitConfig.getGitBranch().startsWith("PR-")){
-            String[] refSpecs = gitSCM.getUserRemoteConfigs().get(0).getRefspec().split("/", 0);
-            gitConfig.setGitBranch(refSpecs[refSpecs.length - 1]);
+            for(String urc: getCleanRefSpecs(gitSCM.getUserRemoteConfigs())) {
```

Comment:
It seems like it would make more sense to move this for loop logic into your `getBranchName` function, and then in there go through the refspecs and return the one you want to the `setGitBranch`

Reply:
I was thinking getBranchName as a function that will take a refSpec and returns the branch name. So the logic to get the refSpec from which we want to get the branch name should be out of this.

Reply:
yeah, I guess my comment is that in line 103 we set the branch name to the default, then we check if it starts with "PR-", and if it does, then we need to make use of the `getUserRemoteConfigs`. And so instead of looping through each "urc" and breaking on the first one that doesn't contain "PR-", I think it would make sense to call a function that does the looping and breaks on the first "valid" name. This way the logic of the function can have some clear unit tests, and if at some point there is an easier way to get the branch name it doesn't get mixed up with the for loop logic used here.